modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/autocomplete/Autocomplete.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAutocomplete from '@clayui/autocomplete';
+import ClayDropDown from '@clayui/drop-down';
+import {FocusScope} from '@clayui/shared';
+import React, {useEffect, useRef, useState} from 'react';
+
+const DefaultItemsWrapperRenderer = ({
+	items,
+	labelKey,
+	onItemClick,
+	valueKey,
+}) => {
+	return (
+		<ClayDropDown.ItemList className="mb-0">
+			{items && items.length === 0 && (
+				<ClayDropDown.Item className="disabled">
+					{Liferay.Language.get('no-items-were-found')}
+				</ClayDropDown.Item>
+			)}
+			{items &&
+				items.length > 0 &&
+				items.map((item) => (
+					<ClayAutocomplete.Item
+						key={String(item[valueKey])}
+						onClick={onItemClick}
+						value={String(item[labelKey])}
+					/>
+				))}
+		</ClayDropDown.ItemList>
+	);
+};
+
+function Autocomplete({
+	disabled = false,
+	id,
+	inputClass,
+	inputId,
+	inputName,
+	inputPlaceholder = Liferay.Language.get('type-here'),
+	inputValue,
+	items,
+	itemsWrapperRenderer = DefaultItemsWrapperRenderer,
+	labelKey,
+	loading,
+	name,
+	onFocus,
+	onInputChange,
+	onSelectedItemChange,
+	required,
+	selectedItem,
+	valueKey,
+}) {
+	const [active, setActive] = useState(false);
+
+	const node = useRef();
+	const dropdownNode = useRef();
+	const inputNode = useRef();
+
+	const currentValue = selectedItem ? selectedItem[valueKey] : null;
+	const currentLabel = selectedItem ? selectedItem[labelKey] : null;
+
+	useEffect(() => {
+		function handleClick(event) {
+			if (
+				node.current.contains(event.target) ||
+				(dropdownNode.current &&
+					dropdownNode.current.contains(event.target))
+			) {
+				return;
+			}
+
+			setActive(false);
+		}
```

Comment:
Is this an implementation for useOnClickOutside?

In ideal world I think We should provide otherProps for ClayAutocomplete.Dropdown because [it is rendering a ClayDropdown.Menu](https://github.com/liferay/clay/blob/master/packages/clay-autocomplete/src/DropDown.tsx#L45) therefore, closeOnClickOutside is an available prop there. 
/cc @matuzalemsteles  @julien

Reply:
Yes, makes sense for me.

Reply:
Can I create a ticket for it?

Reply:
https://github.com/liferay/clay/issues/4284